src/server/verification/processor/EnrollmentProcessor.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import { GunDBPublic } from '../../gun/gun-middleware'
+import AdminWallet from '../../blockchain/AdminWallet'
+
+import { EnrollmentProviders, type EnrollmentProvider, type IEnrollmentEventPayload } from './typings'
+
+import ZoomProvidersFactory from './provider/ZoomProvider'
+import KairosProvidersFactory from './provider/KairosProvider'
+
+class EnrollmentProcessor {
+  static providersFactories = {}
+
+  static registerProviers(factories: { [name: EnrollmentProvider]: Function }) {
+    this.providersFactories = factories
+  }
+
+  user = null
+  storage = null
+  sessionRef = null
+
+  constructor(user, storage, adminApi) {
+    this.user = user
+    this.storage = storage
+    this.adminApi = adminApi
+  }
+
+  validate(payload: any, providerType: EnrollmentProvider = EnrollmentProviders.Kairos) {
+    const { user } = this
+    const identifier = user || {}
+    const { sessionId } = payload || {}
+    const providerInstance = this._createProvider(providerType)
+
+    if (!user || !identifier || !payload || !sessionId || !providerInstance.isPayloadValid(payload)) {
+      throw new Error('Invalid input')
+    }
+  }
+
+  async enroll(payload: any, providerType: EnrollmentProvider = EnrollmentProviders.Kairos) {
+    const { sessionId } = payload
+    const sessionRef = GunDBPublic.session(sessionId)
+    const providerInstance = this._createProvider(providerType)
+
+    this.sessionRef = sessionRef
+
+    try {
+      const enrollmentResult = await providerInstance.enroll(payload, this.user.identifier)
```

Comment:
1. we first search if highly similar face exists (where we use the min match level env variable)
if highly similar face exists then we return failure
2. we use zoomId a special identifier sent by the user from the client and not this.user.identifier
3. if identifier is already enrolled we need to catch this possibility and not return failure, but user already enrolled

Reply:
see old zoom logic and kairos addIfUniqueAndAlive logic